cli/src/cluster_query.rs

```diff
@@ -1077,10 +1105,73 @@ pub fn process_ping(
     timeout: &Duration,
     fixed_blockhash: &Option<Hash>,
     print_timestamp: bool,
+    use_nonce: bool,
+    resubmit_interval: &Duration,
+    blockhash_interval: &Duration,
 ) -> ProcessResult {
     println_name_value("Source Account:", &config.signers[0].pubkey().to_string());
     println!();
 
+    let commitment_for_confirm = match config.commitment.commitment {
+        CommitmentLevel::Root => CommitmentConfig::max(),
+        _ => CommitmentConfig::recent(),
+    };
+
+    let nonce_context = if use_nonce {
+        use solana_sdk::signature::Signer;
+        let nonce_authority = config.signers[0].pubkey();
+        let mut all_nonce_configs = vec![];
+        for i in 0..(count.unwrap() / 5) {
+            let nonce_commitment = CommitmentConfig::root();
+            let nonce_lamports = rpc_client.get_minimum_balance_for_rent_exemption(solana_sdk::nonce::State::size())?;
+            let mut nonce_keypairs = vec![];
+            for _ in 0..5 {
+                nonce_keypairs.push(solana_sdk::signature::Keypair::new());
+            }
+            let mut ixs = vec![];
+            nonce_keypairs.iter().for_each(|nonce_keypair| ixs.extend(solana_sdk::system_instruction::create_nonce_account(
```

Comment:
maybe switch to derived account addresses not to waste bunch of garbage nonce accounts...

Reply:
:+1: Bonus feature in that we can check for their existence at startup and skip creation if so